prog/test/connected_subnets.rb

```diff
@@ -212,5 +212,7 @@ def test_connection(to_connect_ip, connecting, should_fail: false, ipv4: true)
   def start_listening(ipv4: true)
     vm_to_be_connected.sshable.cmd("sudo systemctl stop listening_ipv#{ipv4 ? "6" : "4"}.service")
     vm_to_be_connected.sshable.cmd("sudo systemctl start listening_ipv#{ipv4 ? "4" : "6"}.service")
+
+    nap 1 if vm_to_be_connected.sshable.cmd("sudo systemctl status listening_ipv#{ipv4 ? "4" : "6"}.service").include?("inactive")
```

Comment:
nit: no need to check the same condition 3 times
```suggestion
    service = "listening_ipv#{ipv4 ? "6" : "4"}.service"
    vm_to_be_connected.sshable.cmd("sudo systemctl stop #{service}")
    vm_to_be_connected.sshable.cmd("sudo systemctl start #{service}")

    nap 1 if vm_to_be_connected.sshable.cmd("sudo systemctl status #{service}").include?("inactive")
```